File: searching/binary.rb
#!/bin/ruby
class Array
  def binary_search(val, low=0, high=(length - 1))
    return nil if high < low
    mid = (low + high) / 2
    case
      when self[mid] > val then binary_search(val, low, mid-1)
      when self[mid] < val then binary_search(val, mid+1, high)
      else mid
    end
  end
end

File: searching/binary.py
#!/bin/python
def binary_search(searchMe, key, low = 0, high = -1):
    if(high == -1): high = len(l)-1
    if low == high:
        if searchMe[low] == key: return low
        else: return -1
    mid = (low+high)//2
    if searchMe[mid] > key: return binary_search(searchMe, value, low, mid-1)
    elif searchMe[mid] < key: return binary_search(searchMe, value, mid+1, high)
    else: return mid
